src/stactools/naip/stac.py

```diff
@@ -158,7 +159,7 @@ def create_item(
                         second_xpath, missing_element("File Identifier")
                     )
                 if resource_desc is not None:
-                    dt = str_to_datetime(resource_desc.split("_")[-1])
+                    resource_desc, dt = process_resource_desc(resource_desc)
```

Comment:
the signature of `process_resource_desc` is `Optional[Tuple[str, datetime]]`, so this will fail if it ever returns None, so you need to handle that case.

Reply:
actually, maybe the signature on process_resource_desc doesn't match the domain of values returned by the method?

Reply:
Adjusted function to only process the resource desc. The code for dt won't be duplicated and can be within stac.py